research-software-directory/RSD-as-a-service
# feat: add page that gives overview of project quality

# Add project quality overview page

Changes proposed in this pull request:

* Add a page that lists all projects you maintain and its completeness
* On that page there is a big table which you can sort per column, including a score column that is the amount of columns with a non-zero or non-false value
* As an admin, you can either see projects you maintain or all projects
* Adapt the data-generation script so that some data is not always present

How to test:
* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale data-generation=1`.
* Login as admin and go to http://localhost/user/project-quality
* Since you don't maintain any projects, you should see nothing, so click the switch
* Play around with the table, sort some columns and check on some pages that values in the table correspond to the page
* Become a maintainer of some pages and check that when the switch is turned off, you do see those projects in the table
* Login as a regular user and visit the same page
* You don't get to see the switch, so create some project pages and check again
* Click on you avatar and go to "My {anything}", you should see a link to the project quality page overview

Note, this is a rough implementation, but it should work and be good enough to be usable and to gather feedback on it.

Closes #764

## With added full screen button (bottom-right) 

![image](https://user-images.githubusercontent.com/9204081/227582078-09961a64-27aa-4336-be5a-33704dd8111c.png)

PR Checklist:
